# <u><h1 style="color: red;background-color: yellow;">1)Importing All Libraries We Require</h1><u><br></br>

In [66]:
import numpy as np
import pandas as pd
import re
import requests
from bs4 import BeautifulSoup

import matplotlib.pyplot as plt
import seaborn as sns

import time

from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity = 'all'

### <h1 style="color: blue;background-color: white;">2)Importing csv file by using read_csv </h1><br></br> 

In [67]:
df = pd.read_csv(r'C:\Users\HP\Data science with Python\main_df.csv')

In [68]:
df

Unnamed: 0                       Title Year of Release Duration in min  \
0             0       \n1.\nThe Godfather\n         1972)\n         175 min   
1             1          \n2.\nGoodfellas\n         1990)\n         146 min   
2             2        \n3.\nPulp Fiction\n         1994)\n         154 min   
3             3  \n4.\nThe Usual Suspects\n         1995)\n         106 min   
4             4      \n5.\nApocalypse Now\n         1979)\n         147 min   
..          ...                         ...             ...             ...   
995         995          \n996.\nBlade II\n         2002)\n         117 min   
996         996      \n997.\nFalling Down\n         1993)\n         113 min   
997         997    \n998.\nEyes Wide Shut\n         1999)\n         159 min   
998         998     \n999.\nThe Lost Boys\n         1987)\n          97 min   
999         999      \n1,000.\nDeadpool 2\n         2018)\n         119 min   

                                       Genre  Rating Votes in Views  \
0                 \nCrime, Drama                 9.2        228,221   
1      \nBiography, Crime, Drama                 8.7        379,413   
2                 \nCrime, Drama                 8.9         83,526   
3        \nCrime, Drama, Mystery                 8.5         27,234   
4          \nDrama, Mystery, War                 8.4        407,182   
..                                       ...     ...            ...   
995     \nAction, Horror, Sci-Fi                 6.7        210,883   
996       \nAction, Crime, Drama                 7.6        183,097   
997   \nDrama, Mystery, Thriller                 7.4        324,924   
998             \nComedy, Horror                 7.3        135,899   
999  \nAction, Adventure, Comedy                 7.7        537,177   

    Gross in Millions  Action  Horror  Drama  Crime  Comedy  Thriller  Sci-Fi  \
0             $14.67M       0       0      1      1       0         0       0   
1            $353.01M       0       0      1      1       0         0       0   
2             $10.00M       0       0      1      1       0         0       0   
3              $0.05M       0       0      1      1       0         0       0   
4             $46.89M       0       0      1      0       0         0       0   
..                ...     ...     ...    ...    ...     ...       ...     ...   
995           $82.35M       1       1      0      0       0         0       1   
996           $40.90M       1       0      1      1       0         0       0   
997           $55.69M       0       0      1      0       0         1       0   
998           $32.22M       0       1      0      0       1         0       0   
999          $324.59M       1       0      0      0       1         0       0   

     Mystery  Adventure  Biography  war  
0          0          0          0    0  
1          0          0          1    0  
2          0          0          0    0  
3          1          0          0    0  
4          1          0          0    1  
..       ...        ...        ...  ...  
995        0          0          0    0  
996        0          0          0    0  
997        1          0          0    0  
998        0          0          0    0  
999        0          1          0    0  

[1000 rows x 19 columns]

### <u><h1 style="color: blue;background-color: white;">2)We are removing the Unnecessary items in the data frame</h1><u><br></br>

#### <u><h1 style="color: blue;background-color: white;">i)Data Cleaning</h1><u><br></br>

In [69]:
df['Genre'] = df['Genre'].apply(lambda x:x.replace('\n',''))

In [70]:
df['Duration in min'] = df['Duration in min'].apply(lambda x:x.replace('min',''))

In [71]:
df['Year of Release'] = df['Year of Release'].apply(lambda x:x.replace(')\n',''))

In [72]:
df['Title'] = df['Title'].apply(lambda x:x.replace('\n',''))

In [73]:
df['Votes in Views'] = df['Votes in Views'].apply(lambda x:x.replace(',',''))

In [74]:
df

Unnamed: 0                 Title Year of Release Duration in min  \
0             0       1.The Godfather            1972            175    
1             1          2.Goodfellas            1990            146    
2             2        3.Pulp Fiction            1994            154    
3             3  4.The Usual Suspects            1995            106    
4             4      5.Apocalypse Now            1979            147    
..          ...                   ...             ...             ...   
995         995          996.Blade II            2002            117    
996         996      997.Falling Down            1993            113    
997         997    998.Eyes Wide Shut            1999            159    
998         998     999.The Lost Boys            1987             97    
999         999      1,000.Deadpool 2            2018            119    

                                     Genre  Rating Votes in Views  \
0                 Crime, Drama                 9.2         228221   
1      Biography, Crime, Drama                 8.7         379413   
2                 Crime, Drama                 8.9          83526   
3        Crime, Drama, Mystery                 8.5          27234   
4          Drama, Mystery, War                 8.4         407182   
..                                     ...     ...            ...   
995     Action, Horror, Sci-Fi                 6.7         210883   
996       Action, Crime, Drama                 7.6         183097   
997   Drama, Mystery, Thriller                 7.4         324924   
998             Comedy, Horror                 7.3         135899   
999  Action, Adventure, Comedy                 7.7         537177   

    Gross in Millions  Action  Horror  Drama  Crime  Comedy  Thriller  Sci-Fi  \
0             $14.67M       0       0      1      1       0         0       0   
1            $353.01M       0       0      1      1       0         0       0   
2             $10.00M       0       0      1      1       0         0       0   
3              $0.05M       0       0      1      1       0         0       0   
4             $46.89M       0       0      1      0       0         0       0   
..                ...     ...     ...    ...    ...     ...       ...     ...   
995           $82.35M       1       1      0      0       0         0       1   
996           $40.90M       1       0      1      1       0         0       0   
997           $55.69M       0       0      1      0       0         1       0   
998           $32.22M       0       1      0      0       1         0       0   
999          $324.59M       1       0      0      0       1         0       0   

     Mystery  Adventure  Biography  war  
0          0          0          0    0  
1          0          0          1    0  
2          0          0          0    0  
3          1          0          0    0  
4          1          0          0    1  
..       ...        ...        ...  ...  
995        0          0          0    0  
996        0          0          0    0  
997        1          0          0    0  
998        0          0          0    0  
999        0          1          0    0  

[1000 rows x 19 columns]

#### <u><h1 style="color: blue;background-color: white;">ii)Checking that Nan values are present or not</h1><u><br></br>

In [75]:
df.isna().sum()

Unnamed: 0            0
Title                 0
Year of Release       0
Duration in min       0
Genre                 0
Rating                0
Votes in Views        0
Gross in Millions    60
Action                0
Horror                0
Drama                 0
Crime                 0
Comedy                0
Thriller              0
Sci-Fi                0
Mystery               0
Adventure             0
Biography             0
war                   0
dtype: int64

#### There are 60 nan values in the gross in millions columns

---

### <u><h1 style="color: blue;background-color: white;">iii)Checking the d-types of each column</h1><u><br></br>

In [76]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 19 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Unnamed: 0         1000 non-null   int64  
 1   Title              1000 non-null   object 
 2   Year of Release    1000 non-null   object 
 3   Duration in min    1000 non-null   object 
 4   Genre              1000 non-null   object 
 5   Rating             1000 non-null   float64
 6   Votes in Views     1000 non-null   object 
 7   Gross in Millions  940 non-null    object 
 8   Action             1000 non-null   int64  
 9   Horror             1000 non-null   int64  
 10  Drama              1000 non-null   int64  
 11  Crime              1000 non-null   int64  
 12  Comedy             1000 non-null   int64  
 13  Thriller           1000 non-null   int64  
 14  Sci-Fi             1000 non-null   int64  
 15  Mystery            1000 non-null   int64  
 16  Adventure          1000 n

### <u><h1 style="color: blue;background-color: white;">iv)Dropping the Nan Values</h1><u><br></br>

In [78]:
df.dropna(inplace=True)

In [79]:
df

Unnamed: 0                 Title Year of Release Duration in min  \
0             0       1.The Godfather            1972            175    
1             1          2.Goodfellas            1990            146    
2             2        3.Pulp Fiction            1994            154    
3             3  4.The Usual Suspects            1995            106    
4             4      5.Apocalypse Now            1979            147    
..          ...                   ...             ...             ...   
995         995          996.Blade II            2002            117    
996         996      997.Falling Down            1993            113    
997         997    998.Eyes Wide Shut            1999            159    
998         998     999.The Lost Boys            1987             97    
999         999      1,000.Deadpool 2            2018            119    

                                     Genre  Rating Votes in Views  \
0                 Crime, Drama                 9.2         228221   
1      Biography, Crime, Drama                 8.7         379413   
2                 Crime, Drama                 8.9          83526   
3        Crime, Drama, Mystery                 8.5          27234   
4          Drama, Mystery, War                 8.4         407182   
..                                     ...     ...            ...   
995     Action, Horror, Sci-Fi                 6.7         210883   
996       Action, Crime, Drama                 7.6         183097   
997   Drama, Mystery, Thriller                 7.4         324924   
998             Comedy, Horror                 7.3         135899   
999  Action, Adventure, Comedy                 7.7         537177   

    Gross in Millions  Action  Horror  Drama  Crime  Comedy  Thriller  Sci-Fi  \
0             $14.67M       0       0      1      1       0         0       0   
1            $353.01M       0       0      1      1       0         0       0   
2             $10.00M       0       0      1      1       0         0       0   
3              $0.05M       0       0      1      1       0         0       0   
4             $46.89M       0       0      1      0       0         0       0   
..                ...     ...     ...    ...    ...     ...       ...     ...   
995           $82.35M       1       1      0      0       0         0       1   
996           $40.90M       1       0      1      1       0         0       0   
997           $55.69M       0       0      1      0       0         1       0   
998           $32.22M       0       1      0      0       1         0       0   
999          $324.59M       1       0      0      0       1         0       0   

     Mystery  Adventure  Biography  war  
0          0          0          0    0  
1          0          0          1    0  
2          0          0          0    0  
3          1          0          0    0  
4          1          0          0    1  
..       ...        ...        ...  ...  
995        0          0          0    0  
996        0          0          0    0  
997        1          0          0    0  
998        0          0          0    0  
999        0          1          0    0  

[940 rows x 19 columns]

###  Dropping the Unnamed column

In [80]:
df.drop('Unnamed: 0',axis=1,inplace=True)

In [81]:
df

Title Year of Release Duration in min  \
0         1.The Godfather            1972            175    
1            2.Goodfellas            1990            146    
2          3.Pulp Fiction            1994            154    
3    4.The Usual Suspects            1995            106    
4        5.Apocalypse Now            1979            147    
..                    ...             ...             ...   
995          996.Blade II            2002            117    
996      997.Falling Down            1993            113    
997    998.Eyes Wide Shut            1999            159    
998     999.The Lost Boys            1987             97    
999      1,000.Deadpool 2            2018            119    

                                     Genre  Rating Votes in Views  \
0                 Crime, Drama                 9.2         228221   
1      Biography, Crime, Drama                 8.7         379413   
2                 Crime, Drama                 8.9          83526   
3        Crime, Drama, Mystery                 8.5          27234   
4          Drama, Mystery, War                 8.4         407182   
..                                     ...     ...            ...   
995     Action, Horror, Sci-Fi                 6.7         210883   
996       Action, Crime, Drama                 7.6         183097   
997   Drama, Mystery, Thriller                 7.4         324924   
998             Comedy, Horror                 7.3         135899   
999  Action, Adventure, Comedy                 7.7         537177   

    Gross in Millions  Action  Horror  Drama  Crime  Comedy  Thriller  Sci-Fi  \
0             $14.67M       0       0      1      1       0         0       0   
1            $353.01M       0       0      1      1       0         0       0   
2             $10.00M       0       0      1      1       0         0       0   
3              $0.05M       0       0      1      1       0         0       0   
4             $46.89M       0       0      1      0       0         0       0   
..                ...     ...     ...    ...    ...     ...       ...     ...   
995           $82.35M       1       1      0      0       0         0       1   
996           $40.90M       1       0      1      1       0         0       0   
997           $55.69M       0       0      1      0       0         1       0   
998           $32.22M       0       1      0      0       1         0       0   
999          $324.59M       1       0      0      0       1         0       0   

     Mystery  Adventure  Biography  war  
0          0          0          0    0  
1          0          0          1    0  
2          0          0          0    0  
3          1          0          0    0  
4          1          0          0    1  
..       ...        ...        ...  ...  
995        0          0          0    0  
996        0          0          0    0  
997        1          0          0    0  
998        0          0          0    0  
999        0          1          0    0  

[940 rows x 18 columns]

### <u><h1 style="color: blue;background-color: white;">v)Removing the special characters in Gross in Millions column</h1><u><br></br>

In [82]:
df['Gross in Millions'] = df['Gross in Millions'].apply(lambda x:x.replace('$',''))

In [83]:
df['Gross in Millions'] = df['Gross in Millions'].apply(lambda x:x.replace('M',''))

In [84]:
df

Title Year of Release Duration in min  \
0         1.The Godfather            1972            175    
1            2.Goodfellas            1990            146    
2          3.Pulp Fiction            1994            154    
3    4.The Usual Suspects            1995            106    
4        5.Apocalypse Now            1979            147    
..                    ...             ...             ...   
995          996.Blade II            2002            117    
996      997.Falling Down            1993            113    
997    998.Eyes Wide Shut            1999            159    
998     999.The Lost Boys            1987             97    
999      1,000.Deadpool 2            2018            119    

                                     Genre  Rating Votes in Views  \
0                 Crime, Drama                 9.2         228221   
1      Biography, Crime, Drama                 8.7         379413   
2                 Crime, Drama                 8.9          83526   
3        Crime, Drama, Mystery                 8.5          27234   
4          Drama, Mystery, War                 8.4         407182   
..                                     ...     ...            ...   
995     Action, Horror, Sci-Fi                 6.7         210883   
996       Action, Crime, Drama                 7.6         183097   
997   Drama, Mystery, Thriller                 7.4         324924   
998             Comedy, Horror                 7.3         135899   
999  Action, Adventure, Comedy                 7.7         537177   

    Gross in Millions  Action  Horror  Drama  Crime  Comedy  Thriller  Sci-Fi  \
0               14.67       0       0      1      1       0         0       0   
1              353.01       0       0      1      1       0         0       0   
2               10.00       0       0      1      1       0         0       0   
3                0.05       0       0      1      1       0         0       0   
4               46.89       0       0      1      0       0         0       0   
..                ...     ...     ...    ...    ...     ...       ...     ...   
995             82.35       1       1      0      0       0         0       1   
996             40.90       1       0      1      1       0         0       0   
997             55.69       0       0      1      0       0         1       0   
998             32.22       0       1      0      0       1         0       0   
999            324.59       1       0      0      0       1         0       0   

     Mystery  Adventure  Biography  war  
0          0          0          0    0  
1          0          0          1    0  
2          0          0          0    0  
3          1          0          0    0  
4          1          0          0    1  
..       ...        ...        ...  ...  
995        0          0          0    0  
996        0          0          0    0  
997        1          0          0    0  
998        0          0          0    0  
999        0          1          0    0  

[940 rows x 18 columns]

---

### <u><h1 style="color: blue;background-color: white;">vi)Removing the numbers and pullstop which is present at Title</h1><u><br></br>

## for that we are applying regex function

In [85]:
df['Title'][0]

'1.The Godfather'

In [86]:
import regex as re

p=re.compile(r"\.(.*)")

In [87]:
p.findall(df['Title'][0])

['The Godfather']

In [88]:
d=df['Title'].apply(lambda x:p.findall(x))

In [89]:
d=d.apply(lambda x:"".join(x))

In [90]:
df['Title']=d

In [91]:
df

Title Year of Release Duration in min  \
0         The Godfather            1972            175    
1            Goodfellas            1990            146    
2          Pulp Fiction            1994            154    
3    The Usual Suspects            1995            106    
4        Apocalypse Now            1979            147    
..                  ...             ...             ...   
995            Blade II            2002            117    
996        Falling Down            1993            113    
997      Eyes Wide Shut            1999            159    
998       The Lost Boys            1987             97    
999          Deadpool 2            2018            119    

                                     Genre  Rating Votes in Views  \
0                 Crime, Drama                 9.2         228221   
1      Biography, Crime, Drama                 8.7         379413   
2                 Crime, Drama                 8.9          83526   
3        Crime, Drama, Mystery                 8.5          27234   
4          Drama, Mystery, War                 8.4         407182   
..                                     ...     ...            ...   
995     Action, Horror, Sci-Fi                 6.7         210883   
996       Action, Crime, Drama                 7.6         183097   
997   Drama, Mystery, Thriller                 7.4         324924   
998             Comedy, Horror                 7.3         135899   
999  Action, Adventure, Comedy                 7.7         537177   

    Gross in Millions  Action  Horror  Drama  Crime  Comedy  Thriller  Sci-Fi  \
0               14.67       0       0      1      1       0         0       0   
1              353.01       0       0      1      1       0         0       0   
2               10.00       0       0      1      1       0         0       0   
3                0.05       0       0      1      1       0         0       0   
4               46.89       0       0      1      0       0         0       0   
..                ...     ...     ...    ...    ...     ...       ...     ...   
995             82.35       1       1      0      0       0         0       1   
996             40.90       1       0      1      1       0         0       0   
997             55.69       0       0      1      0       0         1       0   
998             32.22       0       1      0      0       1         0       0   
999            324.59       1       0      0      0       1         0       0   

     Mystery  Adventure  Biography  war  
0          0          0          0    0  
1          0          0          1    0  
2          0          0          0    0  
3          1          0          0    0  
4          1          0          0    1  
..       ...        ...        ...  ...  
995        0          0          0    0  
996        0          0          0    0  
997        1          0          0    0  
998        0          0          0    0  
999        0          1          0    0  

[940 rows x 18 columns]

# Again checking the d-types

In [92]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 940 entries, 0 to 999
Data columns (total 18 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Title              940 non-null    object 
 1   Year of Release    940 non-null    object 
 2   Duration in min    940 non-null    object 
 3   Genre              940 non-null    object 
 4   Rating             940 non-null    float64
 5   Votes in Views     940 non-null    object 
 6   Gross in Millions  940 non-null    object 
 7   Action             940 non-null    int64  
 8   Horror             940 non-null    int64  
 9   Drama              940 non-null    int64  
 10  Crime              940 non-null    int64  
 11  Comedy             940 non-null    int64  
 12  Thriller           940 non-null    int64  
 13  Sci-Fi             940 non-null    int64  
 14  Mystery            940 non-null    int64  
 15  Adventure          940 non-null    int64  
 16  Biography          940 non

### Here we have 'Duration in min','Votes in Views','Gross in Millions' in object, so we have to convert in int.

In [93]:
df['Duration in min']=df["Duration in min"].astype(int)

In [94]:
df['Votes in Views']=df['Votes in Views'].astype(int)

In [95]:
df['Gross in Millions']=df['Gross in Millions'].astype(float)

In [96]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 940 entries, 0 to 999
Data columns (total 18 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Title              940 non-null    object 
 1   Year of Release    940 non-null    object 
 2   Duration in min    940 non-null    int32  
 3   Genre              940 non-null    object 
 4   Rating             940 non-null    float64
 5   Votes in Views     940 non-null    int32  
 6   Gross in Millions  940 non-null    float64
 7   Action             940 non-null    int64  
 8   Horror             940 non-null    int64  
 9   Drama              940 non-null    int64  
 10  Crime              940 non-null    int64  
 11  Comedy             940 non-null    int64  
 12  Thriller           940 non-null    int64  
 13  Sci-Fi             940 non-null    int64  
 14  Mystery            940 non-null    int64  
 15  Adventure          940 non-null    int64  
 16  Biography          940 non

In [97]:
df

Title Year of Release  Duration in min  \
0         The Godfather            1972              175   
1            Goodfellas            1990              146   
2          Pulp Fiction            1994              154   
3    The Usual Suspects            1995              106   
4        Apocalypse Now            1979              147   
..                  ...             ...              ...   
995            Blade II            2002              117   
996        Falling Down            1993              113   
997      Eyes Wide Shut            1999              159   
998       The Lost Boys            1987               97   
999          Deadpool 2            2018              119   

                                     Genre  Rating  Votes in Views  \
0                 Crime, Drama                 9.2          228221   
1      Biography, Crime, Drama                 8.7          379413   
2                 Crime, Drama                 8.9           83526   
3        Crime, Drama, Mystery                 8.5           27234   
4          Drama, Mystery, War                 8.4          407182   
..                                     ...     ...             ...   
995     Action, Horror, Sci-Fi                 6.7          210883   
996       Action, Crime, Drama                 7.6          183097   
997   Drama, Mystery, Thriller                 7.4          324924   
998             Comedy, Horror                 7.3          135899   
999  Action, Adventure, Comedy                 7.7          537177   

     Gross in Millions  Action  Horror  Drama  Crime  Comedy  Thriller  \
0                14.67       0       0      1      1       0         0   
1               353.01       0       0      1      1       0         0   
2                10.00       0       0      1      1       0         0   
3                 0.05       0       0      1      1       0         0   
4                46.89       0       0      1      0       0         0   
..                 ...     ...     ...    ...    ...     ...       ...   
995              82.35       1       1      0      0       0         0   
996              40.90       1       0      1      1       0         0   
997              55.69       0       0      1      0       0         1   
998              32.22       0       1      0      0       1         0   
999             324.59       1       0      0      0       1         0   

     Sci-Fi  Mystery  Adventure  Biography  war  
0         0        0          0          0    0  
1         0        0          0          1    0  
2         0        0          0          0    0  
3         0        1          0          0    0  
4         0        1          0          0    1  
..      ...      ...        ...        ...  ...  
995       1        0          0          0    0  
996       0        0          0          0    0  
997       0        1          0          0    0  
998       0        0          0          0    0  
999       0        0          1          0    0  

[940 rows x 18 columns]

# we have 940 rows but in data frame we have 999 in the last row. we will reset the index.

In [98]:
df.reset_index(inplace=True)

In [99]:
df

index               Title Year of Release  Duration in min  \
0        0       The Godfather            1972              175   
1        1          Goodfellas            1990              146   
2        2        Pulp Fiction            1994              154   
3        3  The Usual Suspects            1995              106   
4        4      Apocalypse Now            1979              147   
..     ...                 ...             ...              ...   
935    995            Blade II            2002              117   
936    996        Falling Down            1993              113   
937    997      Eyes Wide Shut            1999              159   
938    998       The Lost Boys            1987               97   
939    999          Deadpool 2            2018              119   

                                     Genre  Rating  Votes in Views  \
0                 Crime, Drama                 9.2          228221   
1      Biography, Crime, Drama                 8.7          379413   
2                 Crime, Drama                 8.9           83526   
3        Crime, Drama, Mystery                 8.5           27234   
4          Drama, Mystery, War                 8.4          407182   
..                                     ...     ...             ...   
935     Action, Horror, Sci-Fi                 6.7          210883   
936       Action, Crime, Drama                 7.6          183097   
937   Drama, Mystery, Thriller                 7.4          324924   
938             Comedy, Horror                 7.3          135899   
939  Action, Adventure, Comedy                 7.7          537177   

     Gross in Millions  Action  Horror  Drama  Crime  Comedy  Thriller  \
0                14.67       0       0      1      1       0         0   
1               353.01       0       0      1      1       0         0   
2                10.00       0       0      1      1       0         0   
3                 0.05       0       0      1      1       0         0   
4                46.89       0       0      1      0       0         0   
..                 ...     ...     ...    ...    ...     ...       ...   
935              82.35       1       1      0      0       0         0   
936              40.90       1       0      1      1       0         0   
937              55.69       0       0      1      0       0         1   
938              32.22       0       1      0      0       1         0   
939             324.59       1       0      0      0       1         0   

     Sci-Fi  Mystery  Adventure  Biography  war  
0         0        0          0          0    0  
1         0        0          0          1    0  
2         0        0          0          0    0  
3         0        1          0          0    0  
4         0        1          0          0    1  
..      ...      ...        ...        ...  ...  
935       1        0          0          0    0  
936       0        0          0          0    0  
937       0        1          0          0    0  
938       0        0          0          0    0  
939       0        0          1          0    0  

[940 rows x 19 columns]

In [100]:
df.drop('index',axis=1,inplace=True)

In [101]:
df

Title Year of Release  Duration in min  \
0         The Godfather            1972              175   
1            Goodfellas            1990              146   
2          Pulp Fiction            1994              154   
3    The Usual Suspects            1995              106   
4        Apocalypse Now            1979              147   
..                  ...             ...              ...   
935            Blade II            2002              117   
936        Falling Down            1993              113   
937      Eyes Wide Shut            1999              159   
938       The Lost Boys            1987               97   
939          Deadpool 2            2018              119   

                                     Genre  Rating  Votes in Views  \
0                 Crime, Drama                 9.2          228221   
1      Biography, Crime, Drama                 8.7          379413   
2                 Crime, Drama                 8.9           83526   
3        Crime, Drama, Mystery                 8.5           27234   
4          Drama, Mystery, War                 8.4          407182   
..                                     ...     ...             ...   
935     Action, Horror, Sci-Fi                 6.7          210883   
936       Action, Crime, Drama                 7.6          183097   
937   Drama, Mystery, Thriller                 7.4          324924   
938             Comedy, Horror                 7.3          135899   
939  Action, Adventure, Comedy                 7.7          537177   

     Gross in Millions  Action  Horror  Drama  Crime  Comedy  Thriller  \
0                14.67       0       0      1      1       0         0   
1               353.01       0       0      1      1       0         0   
2                10.00       0       0      1      1       0         0   
3                 0.05       0       0      1      1       0         0   
4                46.89       0       0      1      0       0         0   
..                 ...     ...     ...    ...    ...     ...       ...   
935              82.35       1       1      0      0       0         0   
936              40.90       1       0      1      1       0         0   
937              55.69       0       0      1      0       0         1   
938              32.22       0       1      0      0       1         0   
939             324.59       1       0      0      0       1         0   

     Sci-Fi  Mystery  Adventure  Biography  war  
0         0        0          0          0    0  
1         0        0          0          1    0  
2         0        0          0          0    0  
3         0        1          0          0    0  
4         0        1          0          0    1  
..      ...      ...        ...        ...  ...  
935       1        0          0          0    0  
936       0        0          0          0    0  
937       0        1          0          0    0  
938       0        0          0          0    0  
939       0        0          1          0    0  

[940 rows x 18 columns]

# Finally checking the Nan Values are present or not

In [102]:
df.isna().sum()

Title                0
Year of Release      0
Duration in min      0
Genre                0
Rating               0
Votes in Views       0
Gross in Millions    0
Action               0
Horror               0
Drama                0
Crime                0
Comedy               0
Thriller             0
Sci-Fi               0
Mystery              0
Adventure            0
Biography            0
war                  0
dtype: int64

In [103]:
df['Year of Release'].astype(int)

ValueError: invalid literal for int() with base 10: 'I) '

### <u><h1 style="color: blue;background-color: white;">vii)so there are some string values in year of release,we have to clean that too</h1><u><br></br>

In [104]:
df

Title Year of Release  Duration in min  \
0         The Godfather            1972              175   
1            Goodfellas            1990              146   
2          Pulp Fiction            1994              154   
3    The Usual Suspects            1995              106   
4        Apocalypse Now            1979              147   
..                  ...             ...              ...   
935            Blade II            2002              117   
936        Falling Down            1993              113   
937      Eyes Wide Shut            1999              159   
938       The Lost Boys            1987               97   
939          Deadpool 2            2018              119   

                                     Genre  Rating  Votes in Views  \
0                 Crime, Drama                 9.2          228221   
1      Biography, Crime, Drama                 8.7          379413   
2                 Crime, Drama                 8.9           83526   
3        Crime, Drama, Mystery                 8.5           27234   
4          Drama, Mystery, War                 8.4          407182   
..                                     ...     ...             ...   
935     Action, Horror, Sci-Fi                 6.7          210883   
936       Action, Crime, Drama                 7.6          183097   
937   Drama, Mystery, Thriller                 7.4          324924   
938             Comedy, Horror                 7.3          135899   
939  Action, Adventure, Comedy                 7.7          537177   

     Gross in Millions  Action  Horror  Drama  Crime  Comedy  Thriller  \
0                14.67       0       0      1      1       0         0   
1               353.01       0       0      1      1       0         0   
2                10.00       0       0      1      1       0         0   
3                 0.05       0       0      1      1       0         0   
4                46.89       0       0      1      0       0         0   
..                 ...     ...     ...    ...    ...     ...       ...   
935              82.35       1       1      0      0       0         0   
936              40.90       1       0      1      1       0         0   
937              55.69       0       0      1      0       0         1   
938              32.22       0       1      0      0       1         0   
939             324.59       1       0      0      0       1         0   

     Sci-Fi  Mystery  Adventure  Biography  war  
0         0        0          0          0    0  
1         0        0          0          1    0  
2         0        0          0          0    0  
3         0        1          0          0    0  
4         0        1          0          0    1  
..      ...      ...        ...        ...  ...  
935       1        0          0          0    0  
936       0        0          0          0    0  
937       0        1          0          0    0  
938       0        0          0          0    0  
939       0        0          1          0    0  

[940 rows x 18 columns]

In [105]:
df['Year of Release']= df['Year of Release'].astype(int)

ValueError: invalid literal for int() with base 10: 'I) '

In [106]:
df = df[df['Year of Release']!='I) ']

In [107]:
df['Year of Release']= df['Year of Release'].astype(int)

ValueError: invalid literal for int() with base 10: '500) Days of Summer\n'

In [108]:
df = df[df['Year of Release']!='500) Days of Summer\n']

In [109]:
df['Year of Release']= df['Year of Release'].astype(int)

ValueError: invalid literal for int() with base 10: '1985–1998\nEpisode:\nThe Firm\n'

In [110]:
df = df[df['Year of Release']!='1985–1998\nEpisode:\nThe Firm\n']

In [111]:
df['Year of Release']= df['Year of Release'].astype(int)

ValueError: invalid literal for int() with base 10: 'II) '

In [112]:
df = df[df['Year of Release']!='II) ']

In [113]:
df['Year of Release']= df['Year of Release'].astype(int)

ValueError: invalid literal for int() with base 10: 'The Unexpected Virtue of Ignorance'

In [114]:
df = df[df['Year of Release']!='The Unexpected Virtue of Ignorance']

In [115]:
df['Year of Release']= df['Year of Release'].astype(int)

ValueError: invalid literal for int() with base 10: 'III) '

In [116]:
df = df[df['Year of Release']!='III) ']

## we have removed each and every special character and unknown values in year of release column.

### <u><h1 style="color: blue;background-color: white;">viii)Converting Year of Release column into int </h1><u><br></br>

In [117]:
df['Year of Release']= df['Year of Release'].astype(int)

In [118]:
df.info()

# finally we converted the year of release column into int d-type

<class 'pandas.core.frame.DataFrame'>
Int64Index: 906 entries, 0 to 939
Data columns (total 18 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Title              906 non-null    object 
 1   Year of Release    906 non-null    int32  
 2   Duration in min    906 non-null    int32  
 3   Genre              906 non-null    object 
 4   Rating             906 non-null    float64
 5   Votes in Views     906 non-null    int32  
 6   Gross in Millions  906 non-null    float64
 7   Action             906 non-null    int64  
 8   Horror             906 non-null    int64  
 9   Drama              906 non-null    int64  
 10  Crime              906 non-null    int64  
 11  Comedy             906 non-null    int64  
 12  Thriller           906 non-null    int64  
 13  Sci-Fi             906 non-null    int64  
 14  Mystery            906 non-null    int64  
 15  Adventure          906 non-null    int64  
 16  Biography          906 non

In [119]:
df

Title  Year of Release  Duration in min  \
0         The Godfather             1972              175   
1            Goodfellas             1990              146   
2          Pulp Fiction             1994              154   
3    The Usual Suspects             1995              106   
4        Apocalypse Now             1979              147   
..                  ...              ...              ...   
935            Blade II             2002              117   
936        Falling Down             1993              113   
937      Eyes Wide Shut             1999              159   
938       The Lost Boys             1987               97   
939          Deadpool 2             2018              119   

                                     Genre  Rating  Votes in Views  \
0                 Crime, Drama                 9.2          228221   
1      Biography, Crime, Drama                 8.7          379413   
2                 Crime, Drama                 8.9           83526   
3        Crime, Drama, Mystery                 8.5           27234   
4          Drama, Mystery, War                 8.4          407182   
..                                     ...     ...             ...   
935     Action, Horror, Sci-Fi                 6.7          210883   
936       Action, Crime, Drama                 7.6          183097   
937   Drama, Mystery, Thriller                 7.4          324924   
938             Comedy, Horror                 7.3          135899   
939  Action, Adventure, Comedy                 7.7          537177   

     Gross in Millions  Action  Horror  Drama  Crime  Comedy  Thriller  \
0                14.67       0       0      1      1       0         0   
1               353.01       0       0      1      1       0         0   
2                10.00       0       0      1      1       0         0   
3                 0.05       0       0      1      1       0         0   
4                46.89       0       0      1      0       0         0   
..                 ...     ...     ...    ...    ...     ...       ...   
935              82.35       1       1      0      0       0         0   
936              40.90       1       0      1      1       0         0   
937              55.69       0       0      1      0       0         1   
938              32.22       0       1      0      0       1         0   
939             324.59       1       0      0      0       1         0   

     Sci-Fi  Mystery  Adventure  Biography  war  
0         0        0          0          0    0  
1         0        0          0          1    0  
2         0        0          0          0    0  
3         0        1          0          0    0  
4         0        1          0          0    1  
..      ...      ...        ...        ...  ...  
935       1        0          0          0    0  
936       0        0          0          0    0  
937       0        1          0          0    0  
938       0        0          0          0    0  
939       0        0          1          0    0  

[906 rows x 18 columns]

In [120]:
df.reset_index(inplace=True)

In [121]:
df

index               Title  Year of Release  Duration in min  \
0        0       The Godfather             1972              175   
1        1          Goodfellas             1990              146   
2        2        Pulp Fiction             1994              154   
3        3  The Usual Suspects             1995              106   
4        4      Apocalypse Now             1979              147   
..     ...                 ...              ...              ...   
901    935            Blade II             2002              117   
902    936        Falling Down             1993              113   
903    937      Eyes Wide Shut             1999              159   
904    938       The Lost Boys             1987               97   
905    939          Deadpool 2             2018              119   

                                     Genre  Rating  Votes in Views  \
0                 Crime, Drama                 9.2          228221   
1      Biography, Crime, Drama                 8.7          379413   
2                 Crime, Drama                 8.9           83526   
3        Crime, Drama, Mystery                 8.5           27234   
4          Drama, Mystery, War                 8.4          407182   
..                                     ...     ...             ...   
901     Action, Horror, Sci-Fi                 6.7          210883   
902       Action, Crime, Drama                 7.6          183097   
903   Drama, Mystery, Thriller                 7.4          324924   
904             Comedy, Horror                 7.3          135899   
905  Action, Adventure, Comedy                 7.7          537177   

     Gross in Millions  Action  Horror  Drama  Crime  Comedy  Thriller  \
0                14.67       0       0      1      1       0         0   
1               353.01       0       0      1      1       0         0   
2                10.00       0       0      1      1       0         0   
3                 0.05       0       0      1      1       0         0   
4                46.89       0       0      1      0       0         0   
..                 ...     ...     ...    ...    ...     ...       ...   
901              82.35       1       1      0      0       0         0   
902              40.90       1       0      1      1       0         0   
903              55.69       0       0      1      0       0         1   
904              32.22       0       1      0      0       1         0   
905             324.59       1       0      0      0       1         0   

     Sci-Fi  Mystery  Adventure  Biography  war  
0         0        0          0          0    0  
1         0        0          0          1    0  
2         0        0          0          0    0  
3         0        1          0          0    0  
4         0        1          0          0    1  
..      ...      ...        ...        ...  ...  
901       1        0          0          0    0  
902       0        0          0          0    0  
903       0        1          0          0    0  
904       0        0          0          0    0  
905       0        0          1          0    0  

[906 rows x 19 columns]

In [122]:
df.drop('index',axis=1,inplace=True)

In [123]:
df

Title  Year of Release  Duration in min  \
0         The Godfather             1972              175   
1            Goodfellas             1990              146   
2          Pulp Fiction             1994              154   
3    The Usual Suspects             1995              106   
4        Apocalypse Now             1979              147   
..                  ...              ...              ...   
901            Blade II             2002              117   
902        Falling Down             1993              113   
903      Eyes Wide Shut             1999              159   
904       The Lost Boys             1987               97   
905          Deadpool 2             2018              119   

                                     Genre  Rating  Votes in Views  \
0                 Crime, Drama                 9.2          228221   
1      Biography, Crime, Drama                 8.7          379413   
2                 Crime, Drama                 8.9           83526   
3        Crime, Drama, Mystery                 8.5           27234   
4          Drama, Mystery, War                 8.4          407182   
..                                     ...     ...             ...   
901     Action, Horror, Sci-Fi                 6.7          210883   
902       Action, Crime, Drama                 7.6          183097   
903   Drama, Mystery, Thriller                 7.4          324924   
904             Comedy, Horror                 7.3          135899   
905  Action, Adventure, Comedy                 7.7          537177   

     Gross in Millions  Action  Horror  Drama  Crime  Comedy  Thriller  \
0                14.67       0       0      1      1       0         0   
1               353.01       0       0      1      1       0         0   
2                10.00       0       0      1      1       0         0   
3                 0.05       0       0      1      1       0         0   
4                46.89       0       0      1      0       0         0   
..                 ...     ...     ...    ...    ...     ...       ...   
901              82.35       1       1      0      0       0         0   
902              40.90       1       0      1      1       0         0   
903              55.69       0       0      1      0       0         1   
904              32.22       0       1      0      0       1         0   
905             324.59       1       0      0      0       1         0   

     Sci-Fi  Mystery  Adventure  Biography  war  
0         0        0          0          0    0  
1         0        0          0          1    0  
2         0        0          0          0    0  
3         0        1          0          0    0  
4         0        1          0          0    1  
..      ...      ...        ...        ...  ...  
901       1        0          0          0    0  
902       0        0          0          0    0  
903       0        1          0          0    0  
904       0        0          0          0    0  
905       0        0          1          0    0  

[906 rows x 18 columns]

### <u><h1 style="color: blue;background-color: white;">3)Finally we have done cleaning and saving it into csv</h1><u><br></br>

In [124]:
df.to_csv(r'C:\Users\HP\Data science with Python\02) web scrapping\cleaned_df.csv')